debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java

```diff
@@ -254,6 +255,9 @@ else if (oidValue == typeRegistry.geometryArrayOid()) {
                 else if (oidValue == typeRegistry.hstoreOid()) {
                     return hstoreSchema();
                 }
+                else if (oidValue == typeRegistry.ltreeOid()) {
+                    return SchemaBuilder.string().optional();
```

Comment:
Should I use a dedicated schema, such as one with logical name called "io.debezium.data.Ltree"?

Reply:
Yes, please. Similar to hstore where I think we do the same (or JSON). These semantical schemas allow consumers to specifically react to such values, e.g. to derive a corresponding schema in a downstream database.

Reply:
The problem I encountered last weekend was that some schema don't have the semantical schema while the others do. I believe they are stored here: https://github.com/debezium/debezium/tree/master/debezium-core/src/main/java/io/debezium/data

The ones having semantical schemas (excerpt):

- Json
- Uuid
- Xml

The ones without semantical schemas (excerpt):

- Hstore ([declared as map](https://github.com/debezium/debezium/blob/master/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java#L283-L294))
- Citext ([declared as string](https://github.com/debezium/debezium/blob/master/debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java#L248-L250))

What confuse me most is that those without semantical schemas are the ones declared as extensions in [init_database.ddl](https://github.com/debezium/debezium/blob/2a065686bd8df50ae5c427d2847e0008a6b5f256/debezium-connector-postgres/src/test/resources/init_database.ddl). So if I need to align with them, I should not declare a semantical schema.